nightfall-deployer/contracts/Challenges.sol

```diff
@@ -183,7 +186,10 @@ contract Challenges is Stateful, Key_Registry, Config {
             extraPublicInputs.roots[0] = uint256(blockL2ContainingHistoricRoot[0].root);
```

Comment:
This will need to be done as a different challenge because `challengeProofVerification` expects to extract root from an “existing and correct” L2 block, which is then used inside proof verification. In this case, because the L2 block itself does not exist, this challenge can't be used. This needs to write it as a different challenge. Apologies for suggesting to use proof verification challenge





[1:50](https://g-i-t.slack.com/archives/DJW90TUG4/p1662468611267739)
In our case because the L2 block itself does not exist we can’t use that challenge
[1:50](https://g-i-t.slack.com/archives/DJW90TUG4/p1662468618270789)
We need to write it as a different challenge

Reply:
separate challenge logic in Challenge.sol added for historicRoot case